### PR TITLE
[AIRFLOW-5693] Support the "blocks" component for the Slack messages

### PR DESCRIPTION
**Pull request by mingrammer**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: tests/contrib/hooks/test_slack_webhook_hook.py
Comment (ashb) [inline tests/contrib/hooks/test_slack_webhook_hook.py:38]: Does the version of the slack module we support (1.x, not 2.x) support this?

Comment (mingrammer) [reply tests/contrib/hooks/test_slack_webhook_hook.py:38]: It does not use the slack client module, but just HTTP post request.

Comment (ashb) [reply tests/contrib/hooks/test_slack_webhook_hook.py:38]: You make an excelent point :D

**Comment by mingrammer** [comment]
CI failed due to this issue. (https://github.com/docker/for-linux/issues/830)

And, this issue seems to be fixed. Please re-run the CI :) (https://github.com/docker/for-linux/issues/830#issuecomment-543930808)

```
Err:13 https://download.docker.com/linux/debian stretch/stable amd64 Packages
  Writing more data than expected (11116 > 9983)
  Hashes of expected file:
   - Filesize:9983 [weak]
   - SHA512:4faf843dcb2544a9135a36ecfc4b914dd4e9cdada9a5e20e4994f5dd33d2d5aacb44f237bf9bf7048bac571e8be47d04975f8d93a19dab67dbfc231b68012011
   - SHA256:db9d0a054919cfa52440f1d59ec3f4f2fd15b89661c87f35f02a24f3832a6ce6
   - SHA1:d52b8071d1449718bc483c54a5bc698a26107aa6 [weak]
   - MD5Sum:d7b024e64ef3849c725f4ce1f05bf7a7 [weak]
  Release file created at: Sat, 17 Aug 2019 05:04:49 +0000
```

File: airflow/contrib/hooks/slack_webhook_hook.py
Comment (feluelle) [inline airflow/contrib/hooks/slack_webhook_hook.py]: new lines there need to be indented with 4 Spaces.
```suggestion
        dictionaries representing Slack blocks.
```

Comment (mingrammer) [reply airflow/contrib/hooks/slack_webhook_hook.py]: It was intended to align the columns for the second line with the first description line.

with 3 spaces (current)

```python
    :param blocks: The blocks to send on Slack. Should be a list of
                   dictionaries representing Slack blocks.
```

with 4 spaces (suggestion)

```python
    :param blocks: The blocks to send on Slack. Should be a list of
                    dictionaries representing Slack blocks.

# or

    :param blocks: The blocks to send on Slack. Should be a list of
                dictionaries representing Slack blocks.
```

But, there is already 3 space-indented line. (20 + 3 spaces)

> I think this is also for aligning the columns.

```
    :param link_names: Whether or not to find and link channel and usernames in your
                       message
```

It is a little confusing for me. Which rules should I follow?

Thank you.

Comment (feluelle) [reply airflow/contrib/hooks/slack_webhook_hook.py]: It should look like this:
```
    :param blocks: The blocks to send on Slack. Should be a list of
        dictionaries representing Slack blocks.
```

4 spaces starting at the docstring indention (`:param`)

Comment (feluelle) [reply airflow/contrib/hooks/slack_webhook_hook.py]: Yes, `link_names` and `attachments` are also wrongly indented. Can you fix it as well? :)

Comment (mingrammer) [reply airflow/contrib/hooks/slack_webhook_hook.py]: I got it. I'll fix that!

Comment (mingrammer) [reply airflow/contrib/hooks/slack_webhook_hook.py]: Sure :)

Comment (mingrammer) [reply airflow/contrib/hooks/slack_webhook_hook.py]: I'm done.

Comment (feluelle) [reply airflow/contrib/hooks/slack_webhook_hook.py]: Thanks :)

**Comment by feluelle** [comment]
@mingrammer do you mind fixing another doc issue in it as well? 😀 

See: https://airflow.readthedocs.io/en/latest/_api/airflow/contrib/hooks/slack_webhook_hook/index.html#airflow.contrib.hooks.slack_webhook_hook.SlackWebhookHook._build_slack_message

There is missing a blank line between the title and the param section in the `_build_slack_message`.

And this:
```diff
- :return: Slack message (str) to send
+ :return: Slack message to send
+ :rtype: str
```

After that I am happy to merge it when everything is green :)

**Comment by mingrammer** [comment]
I fixed all docstring issues in this file!

**Review by feluelle — APPROVED** [review]
Nice!

**Pull request merged**
